tasks/image/pull.go

```diff
@@ -13,8 +13,22 @@ import (
 func RunPull(ctx *context.ExecuteContext, t *Task, _ bool) (bool, error) {
 	record, err := getImageRecord(recordPath(ctx, t.config))
 	if err != nil {
-		t.logger().Warnf("Failed to get image record: %s", err)
+		if os.IsNotExist(err) {
+			t.logger().Info("Creating image record")
+			image, err := GetImage(ctx, t.config)
+			if err != nil {
+				return false, err
+			}
+			record := imageModifiedRecord{ImageID: image.ID}
+			updateImageRecord(recordPath(ctx, t.config), record)
+			if err != nil {
+				return false, err
+			}
+		} else {
+			t.logger().Warnf("Failed to get image record: %s", err)
+		}
```

Comment:
I think this change is likely part of what is breaking the example tests. This shouldn't be necessary because the line below `t.config.Pull.required()` should always return false (because `pullLocal`  returns false) and `RunPull` will exit there.

Does it work if you remove this block?

Reply:
This section was intended to create the dobi record file for the local present image. 
Without the section the image is used properly, but the warning could disturb the force.
```
[WARN] [image:pull runtime] imagename Failed to get image record: stat /home/user/dobi-dev-test/.dobi/images/imagename latest: no such file or directory
```

I just replaced this section and tests are running fine now. Do you have any idea what i've done wrong here?

Reply:
The test output is stored as an artifact in CI: https://app.circleci.com/jobs/github/dnephin/dobi/1176/artifacts

It looks like both failures are a result of `failed to execute task "runner:pull": no such image`.

This change would run any time an image record is missing, not only when `pull: never` is set. 

I think the call to `GetImage` is the problem, but I haven't dug into why exactly.

Reply:
> Without the section the image is used properly, but the warning might disturb the force for some users.

Good thinking!

Something like this might work, but if not I think the warning is fine for now:

```go
record, err := getImageRecord(recordPath(ctx, t.config))
switch {
case !t.config.Pull.Required(record.LastPull):
    t.logger().Debugf("Pull not required")
    return false, nil
case err != nil:
    t.logger().Warnf("Failed to get image record: %s", err)
}
```

Reply:
This seems to work now for the 'never' use case, but gives on a regular 'once' or 'always' the "Failed to get image record" error message. However it proceeds and pulls the image in both cases.
I guess it would be better to restore the original case for 'once' and 'always' as i see no obious way to get the value of the 'pull' key to check wether it was 'never' or not.

Reply:
Yup, exactly! That warning is expected the first time since the record is missing.